src/parser/smt2/smt2_term_parser.cpp

```diff
@@ -0,0 +1,168 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2022 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Term parser for smt2
+ */
+
+#include "parser/smt2/smt2_term_parser.h"
+
+#include <string.h>
+
+#include "base/check.h"
+#include "base/output.h"
+
+namespace cvc5 {
+namespace parser {
+
+Smt2TermParser::Smt2TermParser(Smt2Lexer& lex, Smt2State& state)
+    : d_lex(lex), d_state(state)
+{
+}
+
+std::string Smt2TermParser::parseSymbol(DeclarationCheck check, SymbolType type)
+{
+  Token tok = d_lex.nextToken();
+  std::string id = tokenStrToSymbol(tok);
+  // run the check
+  if (!d_state.isAbstractValue(id))
+  {
+    // if an abstract value, SolverEngine handles declaration
+    d_state.checkDeclaration(id, check, type);
+  }
+  return id;
+}
+
+std::vector<std::string> Smt2TermParser::parseSymbolList(DeclarationCheck check,
+                                                         SymbolType type)
+{
+  d_lex.eatToken(Token::LPAREN_TOK);
+  std::vector<std::string> symbols;
+  Token tok = d_lex.nextToken();
+  while (tok != Token::RPAREN_TOK)
+  {
+    d_lex.reinsertToken(tok);
+    std::string sym = parseSymbol(check, type);
+    symbols.push_back(sym);
+    tok = d_lex.nextToken();
+  }
+  return symbols;
+}
+
+std::string Smt2TermParser::parseKeyword()
+{
+  d_lex.eatToken(Token::KEYWORD);
+  std::string s = d_lex.tokenStr();
+  // strip off the initial colon
+  return s.substr(1);
```

Comment:
Could you avoid the extra allocation by using [erase](https://cplusplus.com/reference/string/string/erase/) (form (1))?